src/Query.php

```diff
@@ -46,14 +46,16 @@
  *     track_total_hits?: bool|int,
  *     version?: bool,
  * }
- * @phpstan-type TSortArgs = list<non-empty-string|array<string, string>|array<string, array{
+ * @phpstan-type TSortArgs = list<TSortArg>|TSortArrayArg
+ * @phpstan-type TSortArg = non-empty-string|TSortArrayArg
+ * @phpstan-type TSortArrayArg = array<string, string>|array<string, array{
  *     order?: non-empty-string,
  *     mode?: non-empty-string,
  *     numeric_type?: non-empty-string,
  *     nested?: array{path: non-empty-string, filter?: array<mixed>, max_children?: int, nested?: array<mixed>},
  *     missing?: non-empty-string,
  *     unmapped_type?: non-empty-string,
- * }>|array{_geo_distance: array<string, mixed>}>
+ * }>|array{_geo_distance: array<string, mixed>}
```

Comment:
Here the thing is that this line fails:

https://github.com/ruflin/Elastica/blob/5fc25a4966b48ae9a631e345b167c867250a43c7/tests/QueryTest.php#L118

the type was correct since it should be a list of arrays like in 

https://github.com/ruflin/Elastica/blob/5fc25a4966b48ae9a631e345b167c867250a43c7/tests/QueryTest.php#L106

but in the case of line 118 it is an array and apparently it works fine, so I've split the type into `TSortArg` which can be used in `addSort` (that expects a mixed) and `TSortArrayArg` to be able to pass just an array to `setSort` like in line 118 of `QueryTest`